### ISSUE TITLE
loss Inf

### ISSUE THREAD

**Issue by hdzmtsssw** [redrock303/NeRFLiX_CPVR2023#6]
Hi, thanks for the great work.
I have attempted to train the model from scratch using the **default settings** in the repository. However, at around 7700 iterations, the loss began to increase and eventually reached`Inf`. Do you have any insight into this issue?
Additionally, I noticed that during training, you randomly select 3 frames from the same scene in LLFF-T, instead of using the view selection strategy as described in the paper. This appears to be inconsistent with the paper, so I am a bit confused.

Here's the stack trace:
```
23-06-27 17:32:52.138 - INFO: [Val Result] Iter: 7600, PSNR: 36.2488, SSIM: 0.9484                                                                                                                                                                                   
23-06-27 17:33:01.007 - INFO: Iter: 7610, LR: 4.996e-04, loss: 0.0193,                                                                                                                                                                                               
23-06-27 17:33:09.883 - INFO: Iter: 7620, LR: 4.996e-04, loss: 0.0280,                                                                                                                                                                                               
23-06-27 17:33:18.755 - INFO: Iter: 7630, LR: 4.996e-04, loss: 0.0230,                                                                                                                                                                                               
23-06-27 17:33:27.622 - INFO: Iter: 7640, LR: 4.996e-04, loss: 0.0336,                                                                                                                                                                                               
23-06-27 17:33:36.491 - INFO: Iter: 7650, LR: 4.996e-04, loss: 0.0119,                                                                                                                                                                                               
23-06-27 17:33:36.492 - INFO: [Validating] Iter: 7650                                                                                                                                                                                                                
val_loader 7824                                                                                                                                                                                                                                                      
33.83551946113983                                                                                                                 
23-06-27 17:33:40.921 - INFO: [Val Result] Iter: 7650, PSNR: 33.8355, SSIM: 0.9149
23-06-27 17:33:49.800 - INFO: Iter: 7660, LR: 4.996e-04, loss: 0.0084, 
23-06-27 17:33:58.683 - INFO: Iter: 7670, LR: 4.996e-04, loss: 0.0246, 
23-06-27 17:34:07.552 - INFO: Iter: 7680, LR: 4.996e-04, loss: 0.0307, 
23-06-27 17:34:16.425 - INFO: Iter: 7690, LR: 4.996e-04, loss: 0.0115, 
23-06-27 17:34:25.298 - INFO: Iter: 7700, LR: 4.995e-04, loss: 0.0157, 
23-06-27 17:34:25.298 - INFO: [Validating] Iter: 7700
val_loader 7824   
34.254843773868025                                                                                                                
23-06-27 17:34:29.822 - INFO: [Val Result] Iter: 7700, PSNR: 34.2548, SSIM: 0.9268
23-06-27 17:34:38.706 - INFO: Iter: 7710, LR: 4.995e-04, loss: 84.0570,      
23-06-27 17:34:47.576 - INFO: Iter: 7720, LR: 4.995e-04, loss: 3081528.7500,    
23-06-27 17:34:56.441 - INFO: Iter: 7730, LR: 4.995e-04, loss: 3187832576.0000,   
23-06-27 17:35:05.312 - INFO: Iter: 7740, LR: 4.995e-04, loss: 780434866176.0000, 
23-06-27 17:35:14.182 - INFO: Iter: 7750, LR: 4.995e-04, loss: 4633922048.0000, 
23-06-27 17:35:14.183 - INFO: [Validating] Iter: 7750
val_loader 7824   
4.9598057312293555                                                                                                                
23-06-27 17:35:19.377 - INFO: [Val Result] Iter: 7750, PSNR: 4.9598, SSIM: 0.0200     
23-06-27 17:35:28.246 - INFO: Iter: 7760, LR: 4.995e-04, loss: 3203675161886720.0000, 
23-06-27 17:35:37.118 - INFO: Iter: 7770, LR: 4.995e-04, loss: 8677076500480.0000, 
WARNING:root:NaN or Inf found in input tensor.                                                                                    
23-06-27 17:35:45.985 - INFO: Iter: 7780, LR: 4.995e-04, loss: inf, 
INFO:base:Iter: 7780, LR: 4.995e-04, loss: inf, 
WARNING:root:NaN or Inf found in input tensor.                                                                                    
23-06-27 17:35:54.845 - INFO: Iter: 7790, LR: 4.995e-04, loss: inf,                                                                                                                                                                                                  
INFO:base:Iter: 7790, LR: 4.995e-04, loss: inf,                                                                                                                                                                                                                      
WARNING:root:NaN or Inf found in input tensor.                                                                                                                                                                                                                       
23-06-27 17:36:03.703 - INFO: Iter: 7800, LR: 4.995e-04, loss: inf,                                                                                                                                                                                                  
INFO:base:Iter: 7800, LR: 4.995e-04, loss: inf,                                                                                                                                                                                                                      
23-06-27 17:36:03.703 - INFO: [Validating] Iter: 7800
INFO:base:[Validating] Iter: 7800
val_loader 7824
7.648982134820619
23-06-27 17:36:08.328 - INFO: [Val Result] Iter: 7800, PSNR: 7.6490, SSIM: 0.4575
INFO:base:[Val Result] Iter: 7800, PSNR: 7.6490, SSIM: 0.4575
WARNING:root:NaN or Inf found in input tensor.
```
![7600_cat_check](https://github.com/redrock303/NeRFLiX_CVPR2023/assets/58366842/34a0c857-1838-4671-bd97-b68492331d09)
> 7600_cat_check⬆️

![8000_cat_check](https://github.com/redrock303/NeRFLiX_CVPR2023/assets/58366842/90adf553-e7f4-4bde-832e-afc073f82f76)
> 8000_cat_check⬆️

**Comment by redrock303** [comment]
Training DCNs can be challenging, and while I have not encountered this specific problem myself, it may be possible to prevent an infinite loss by using a lower learning rate.  Additionally, we offer this simplified version that provides more paired training samples. However, you could also attempt to take only the two closely matched views for reference.

**Comment by hdzmtsssw** [comment]
Thanks for your fast reply. I attempted to load the released model and finetune using the default settings, but the loss became NaN at around 10500 iterations. I plan to try training with a lower learning rate to see if that resolves the issue.

You mentioned that you offer this **simplified version**, so I would like to confirm whether you have a **complete version** that uses two closely matched views for reference. Which version did you use to train the model that you released, and which one is expected to achieve better performance? I believe that the simplified version does not provide additional paired training samples, as both the simplified and the complete versions select one rendered image and two reference images.

**Comment by redrock303** [comment]
> 

1. Using fixed-matched images as a reference provides only a single training sample, such as I0 I1. On the other hand, the random selection strategy generates multiple possible training samples (a target view combined with different reference views), such as (i0, i1), (i0, i2), and so on. This strategy can be considered a form of data augmentation, as it increases the number of training samples. Since the LLFF data is forward-facing, it is relatively easy to estimate correspondences between randomly selected views and the target view. However, during the inference phase, we suggest using a view selection scheme to ensure optimal performance. 
2. The model that has been made available to the public has been retrained using a simplified version, which has yielded better quantitative results than what was reported in our research paper on the Tanks and Temples benchmark.

**Comment by GuangYuanKK** [comment]
@hdzmtsssw Hello, I was wondering if you have found a solution of the unexpected loss (inf or nan)?

**Comment by hdzmtsssw** [comment]
> @hdzmtsssw Hello, I was wondering if you have found a solution of the unexpected loss (inf or nan)?

Hi, I resolved this issue by adjusting the `BASE_LR` from 5e-4 to 2e-4, but it should be noted that the results I obtained seem to be problematic. Whether it is training from scratch or fine-tuning, the results are very poor. Please refer to this link for more details: https://github.com/redrock303/NeRFLiX_CVPR2023/issues/7
